src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -954,7 +954,7 @@ void CodeGen::genCodeForBinary(GenTreeOp* treeNode)
     // reg3 = reg3 op reg2
     else
     {
-        inst_RV_RV(ins_Copy(targetType), targetReg, op1reg, targetType);
+        inst_RV_RV(ins_Copy(targetType), targetReg, op1reg, op1->TypeGet());
```

Comment:
Hmm, should `gcMarkRegPtrVal` below also use `op1->TypeGet()`?

Reply:
AFAICT, the codegen-time gc info is used at labels and calls, and hence, since this is a transitory byref, it wouldn't show up as being wrong in this case. That said, I believe it should be maintained correctly.